### PR TITLE
feat: task due date warning badge (#180)

### LINKED ISSUE #180: [OpenSpec] [planix] task-due-date-warning
Add visual warning badges to task cards on the kanban board, indicating when tasks are approaching their due date (within 2 days) or overdue. This gives users immediate visual feedback about deadline urgency without opening individual tasks.

## Specs

- **kanban-board** — adds Due Date Status Helper and Due Date Badge on Task Card requirements (2 new requirements, 10 scenarios)

### Frontend Implementation

- [ ] **1. Add `dueDateStatus` computed helper** (`src/utils/taskHelpers.js`)
  - [ ] Takes a task object, returns `null` | `"approaching"` | `"overdue"` based on `dueDate` vs today
  - [ ] Uses date-only comparison (no time component)

- [ ] **2. Add due date badge to `TaskCard.vue`**
  - [ ] Yellow chip "Due soon" when approaching (within 2 days)
  - [ ] Red chip "Overdue" when past due
  - [ ] No chip otherwise

- [ ] **3. Add CSS styling for badge states**
  - [ ] Uses Nextcloud theming variables for colors

- [ ] **4. Add unit test for `dueDateStatus` helper**
  - [ ] No due date → null
  - [ ] Future date (>2 days) → null
  - [ ] Approaching date (1-2 days) → "approaching"
  - [ ] Today → "approaching"
  - [ ] Past date → "overdue"

### PR DESCRIPTION
**Pull request by WilcoLouwerse**
Closes #180

## Summary
Adds visual warning indicators to tasks that are approaching or past their due date. Tasks due within 2 days display a yellow "Due soon" badge, and overdue tasks display a red "Overdue" badge. The implementation includes a helper function for computing due date status and unit tests covering all boundary conditions.

## Spec Reference
- Issue: #180
- Spec: `openspec/changes/task-due-date-warning/design.md`

## Changes
- `src/utils/taskHelpers.js` — Added `dueDateStatus` helper that computes task urgency (approaching/overdue/null) based on due date
- `src/components/TaskCard.vue` — New task card component displaying task details with due date warning badges
- `src/utils/__tests__/taskHelpers.test.js` — Unit tests covering no due date, future dates, approaching dates, and overdue dates
- `vitest.config.js` — Test configuration for the project
- `package.json` — Added vitest testing framework and dependencies
- `l10n/en.json` — English translations for badge text and status labels
- `l10n/nl.json` — Dutch translations for badge text and status labels

## Test Coverage
- `src/utils/__tests__/taskHelpers.test.js` — Comprehensive test suite with 15 tests covering:
  - No due date handling (null/undefined)
  - Future dates (>2 days away)
  - Approaching dates (0-2 days)
  - Overdue dates (past due)
  - Date parsing and time component handling

All tests pass successfully. The implementation also passes linting and builds without errors.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,12 +50,15 @@
...
-    "eslint-plugin-vue": "^9.21.1",
-    "vue-eslint-parser": "^9.4.3",
     "eslint": "^8.56.0",
     "eslint-import-resolver-alias": "^1.1.2",
+    "eslint-plugin-vue": "^9.21.1",
+    "happy-dom": "^20.9.0",
+    "jsdom": "^29.0.2",
     "style-loader": "~4.0.0",
     "stylelint": "^15.11.0",
+    "vitest": "^1.0.0",
```
Comment (rubenvdlinde) [inline package.json:64]: [unfixed: vitest@4.x upgrade is a breaking change requiring separate validation] Rule: GHSA-67mh-4wv8-2f99 / OWASP A06:2021 — `vitest@^1.0.0` introduces a vulnerable `esbuild <=0.24.2` via the `vite`/`vite-node` chain. The esbuild vulnerability allows any website to send requests to the development server and read responses. Risk is limited to developer workstations during test runs with `vitest --ui`; normal `vitest run` in CI does not expose a public server. Fix: upgrade to `vitest@4.x` (`npm audit fix --force`), but this is a semver-breaking change — `npm audit` itself warns it would install `vitest@4.1.4`. Escalated for product-team validation.

**Comment by rubenvdlinde** [comment]
## Security Review — Clyde Barcode

**Result: FAIL** (0 fixed, 1 unfixed WARNING, 0 CRITICAL)

### Checks Run
- [x] `semgrep scan --config=p/security-audit --config=p/owasp-top-ten --metrics=off` — 0 findings (7 files, 89 rules)
- [x] `semgrep scan --config=p/secrets --metrics=off` — 0 findings
- [x] `gitleaks detect --no-git -s .` — no leaks found
- [x] `npm audit --production` — 13 pre-existing vulnerabilities (9 low, 4 moderate); none introduced by this PR
- [x] `npm audit` (full, including devDeps) — 1 new WARNING: vitest esbuild chain (GHSA-67mh-4wv8-2f99)
- [x] `composer audit` — no PHP packages to audit (skipped; repo has no PHP dependencies in lock)
- [x] Manual OWASP Top 10 diff review — no `v-html`, no `eval()`, no `innerHTML`, no hardcoded credentials, Vue auto-escaping used correctly throughout

### Findings

| Severity | File | Rule | Status |
|----------|------|------|--------|
| WARNING | `package.json:60` | GHSA-67mh-4wv8-2f99 / OWASP A06:2021 | [unfixed: breaking change] |

### Notes

**New code (TaskCard.vue, taskHelpers.js):** Clean. Vue template auto-escaping applied correctly, no `v-html`, no XSS vectors. The `t()` i18n function handles parameterized strings safely. The `dueDateStatus` helper performs pure date arithmetic with no security-sensitive operations.

**Pre-existing npm audit findings** (axios SSRF, dompurify bypass, follow-redirects, vue ReDoS, brace-expansion DoS) are out of scope for this PR — they were present on `main` before this branch.

**Introduced WARNING:** `vitest@^1.0.0` (new devDependency) pulls in `vite` → `esbuild <=0.24.2` (GHSA-67mh-4wv8-2f99). The vulnerability allows any website to send requests to the esbuild/vite dev server. Impact is limited to developer workstations when running `vitest --ui`; standard CI `vitest run` does not expose a public server. Fix requires `vitest@4.x` (breaking change per `npm audit` output). See inline comment for details.

**Verdict:** FAIL — one unfixed WARNING in new devDependency chain. Applier (Axel Pliér) to decide whether to block or accept the dev-only risk.

See inline comment for per-finding detail.

**Comment by rubenvdlinde** [comment]
Heads-up from the Vue 3 migration: #292 (`feat/vue3-migration` → `development`) rewrites nearly every `.vue` file, including `TaskCard.vue` which this draft touches.

This PR is already `CONFLICTING` against `development` and is still a draft, so nothing is lost — but when you pick it up, rebase **after** #292 lands. Relevant to the due-date badge specifically:

- `NcChip`'s `type` prop is now `variant`, and the value `'default'` was never valid in either major — `dueDateBadgeType` is now `dueDateBadgeVariant` and maps the neutral case to `'secondary'` (NcChip's real default).
- `tests/e2e/kanban-board.spec.ts` now opens the **seeded** fixture project via `tests/e2e/nav.ts` instead of clicking `a[href*="/projects/"]`, which matched nothing (planix renders no project anchors) — so those badge assertions actually execute now.
